### PR TITLE
[SQDEV-3406] Support ability to render widgets without auth information

### PR DESCRIPTION
**Pull request by Locrian24**
## Description of the change

> Replaces the `squatch.widgets().render` function away from deprecated `cookieUser` to instead call the new graphql endpoint which supports missing `user` and `jwt` properties to be passed into `squatch.init`.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3406
- Process.st launch checklist: https://app.process.st/runs/Instant-Access-Widgets-Frontend-vJNCQUjcR8wxqIxSPVlPBA

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist

### Code review

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: src/squatch.ts [hunk 1]
```diff
@@ -67,6 +68,32 @@ export function events(): EventsApi | null {
...
+ * Entry-point for high level API to render a widget using the instance of {@link Widgets} created when you call {@link #init init}.
+ */
+export function widget(
+  widgetConfig: WidgetConfig
+): Promise<WidgetResult | undefined> | undefined {
+  return widgets()?.render(widgetConfig);
+}
+
+/**
+ * Initial concept for automatic widget rendering
```
Comment (johanventer) [inline src/squatch.ts:81]: This doc comment could be better

**Review by johanventer — COMMENTED** [review]
Looks pretty good

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.5.0] - 2022-12-01
```
Comment (00salmon) [inline CHANGELOG.md:10]: flagging changelog date

File: demo/index.html [hunk 3]
```diff
@@ -61,29 +61,45 @@
...
+      //     setStatus("Done in " + (Date.now() - start) / 1000 + " seconds");
+      //     hideSpineer();
+      //     if (res.widget.open) res.widget.open();
+      //   })
+      //   .catch(function (err) {
+      //     console.log("Upsert error", err);
+      //     setStatus("Error" + JSON.stringify(err));
+      //     hideSpineer();
+      //   });
+      // });
```
Comment (00salmon) [inline demo/index.html:102]: how does this affect how squath-js works if this were to be merged and run later?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/squatch-js",
-  "version": "2.4.3",
+  "version": "2.4.3-5",
```
Comment (00salmon) [inline package.json:3]:
```suggestion
  "version": "2.5.0",
```

File: package.json [hunk 5]
```diff
@@ -76,7 +76,7 @@
     "npm-run-all": "^4.1.5",
     "parcel": "^1.12.3",
     "parcel-bundler": "^1.12.4",
-    "playwright": "^1.0.2",
+    "playwright": "^1.33.0-alpha-mar-29-2023",
```
Comment (00salmon) [inline package.json:79]: any reason for this very specific version? 👀

File: src/api/WidgetApi.ts [hunk 6]
```diff
@@ -114,17 +101,52 @@ export default class WidgetApi {
...
+            id
+            accountId
+          }
+          jsOptions
+          widgetConfig {
+            values
+          }
+        }
+      }
+    `;
```
Comment (00salmon) [inline src/api/WidgetApi.ts:128]: can we move this somewhere else? i imagine we'll be adding more queries soon and having them in the function gets a bit cluttered

File: src/utils/utmUtils.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+      ...(configIn ? { configIn } : {}),
+      domain,
+      tenantAlias,
+    },
+  };
+}
+
+/**
+ * Converts _saasquatchExtra into
+ * @param obj
```
Comment (00salmon) [inline src/utils/utmUtils.ts:54]: do we have an actual type definition for this?

File: src/widgets/PopupWidget.ts [hunk 8]
```diff
@@ -54,6 +54,11 @@ export default class PopupWidget extends Widget {
       "display: none; position: fixed; z-index: 1; padding-top: 5%; left: 0; top: -2000px; width: 100%; height: 100%; overflow: auto; background-color: rgba(0,0,0,0.4);"
     );
 
+    document.head.insertAdjacentHTML(
+      "beforeend",
+      `<style>#squatchModal::-webkit-scrollbar { display: none; }</style>`
+    );
+
```
Comment (00salmon) [inline src/widgets/PopupWidget.ts:61]: @johanventer Zach added this to fix a double-scrollbar bug, could this cause any issues? 👀

File: src/widgets/Widget.ts [hunk 9]
```diff
@@ -225,8 +225,10 @@ export default abstract class Widget {
         email: email || null,
         firstName: firstName || null,
         lastName: lastName || null,
-        id: this.context.user.id,
-        accountId: this.context.user.accountId,
+
+        // FIXME: Double check this
+        id: this.context.user!.id,
+        accountId: this.context.user!.accountId,
```
Comment (00salmon) [inline src/widgets/Widget.ts:231]: flagging FIXME

File: CHANGELOG.md [hunk 10]
```diff
@@ -7,6 +7,40 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.5.0] - 2023-05-15
```
Comment (00salmon) [inline CHANGELOG.md:10]:
```suggestion
## [2.5.0] - 2023-05-16
```

**Pull request merged**